### PR TITLE
enosys --dump and setpriv --seccomp

### PR DESCRIPTION
**Pull request by t-8ch**
Note: This is for after 2.40 has been released.

### CODE REVIEW

**Comment by karelzak** [comment]
For v2.41, right?

This time, I won't be creating a `next` branch. I think it's confusing. It seems better to create stable/v2.40 where we can continue to work on a stable release and keep the `master` branch as usual for PRs.

**Comment by t-8ch** [comment]
> For v2.41, right?

Yes.

IMO we can also keep the PR open and add a label.
Depending on the length the RC phase.

**Comment by karelzak** [comment]
It's better to merge new code to `master` and force people to use/test it as soon as possible :-)

File: include/seccomp.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/*
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ * Copyright (C) 2023 Thomas Weißschuh <thomas@t-8ch.de>
```
Comment (karelzak) [inline include/seccomp.h:4]: It's already February :-)

File: sys-utils/setpriv.1.adoc [hunk 2]
```diff
@@ -110,6 +110,10 @@ For example grant file read access to everything under */boot*:
 +
 *--landlock-rule path-beneath:read-file:/boot*
 
+*--seccomp-filter* _fle_::
+
+Load seccomp filter from file.
```
Comment (karelzak) [inline sys-utils/setpriv.1.adoc:115]: Please, more words about the file.

File: misc-utils/enosys.1.adoc [hunk 3]
```diff
@@ -36,6 +36,9 @@ List syscalls known to *enosys*.
 *-m*, *--list-ioctl*::
 List ioctls known to *enosys*.
 
+*-d*, *--dump*::
+Dump seccomp bytecode filter to standard output.
```
Comment (karelzak) [inline misc-utils/enosys.1.adoc:40]: Add hint that the file is usable with setpriv.

File: misc-utils/enosys.c [hunk 4]
```diff
@@ -244,6 +239,12 @@ int main(int argc, char **argv)
 
 	INSTR(BPF_STMT(BPF_RET | BPF_K, SECCOMP_RET_ALLOW));
 
+	if (dump) {
+		if (write_all(STDOUT_FILENO, filter, (f - filter) * sizeof(filter[0])))
+			err(EXIT_FAILURE, _("Could not dump seccomp filter"));
+		return EXIT_SUCCESS;
```
Comment (karelzak) [inline misc-utils/enosys.c:245]: How portable is the filter? (I mean archs, kernel versions, etc.)

Comment (t-8ch) [reply misc-utils/enosys.c:245]: Archs, not at all. The script will detect the mismatch and kill the process.

Every other dimension should be portable.

**Pull request merged**